internal/resources/iampolicy/scope_helper.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	iammodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+)
+
+type scopeType int
+
+const (
+	unknown scopeType = iota
+	organization
+	clusterGroup
+	cluster
+	workspace
+	namespace
+)
+
+func stringToScope(s string) scopeType {
+	switch s {
+	case organizationKey:
+		return organization
+	case clusterGroupKey:
+		return clusterGroup
+	case clusterKey:
+		return cluster
+	case workspaceKey:
+		return workspace
+	case namespaceKey:
+		return namespace
+	default:
+		return unknown
+	}
+}
+
+func (s scopeType) String() string {
+	return [...]string{
+		"unknown",
+		organizationKey,
+		clusterGroupKey,
+		clusterKey,
+		workspaceKey,
+		namespaceKey,
+	}[s]
+}
+
+func validateScope(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(scopeKey)
+	if !ok {
+		return fmt.Errorf("scope: %v is not valid: minimum one valid scope block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("scope data: %v is not valid: minimum one valid scope block is required", data)
+	}
+
+	scopeData := data[0].(map[string]interface{})
+	scopesFound := make([]string, 0)
+	scopesAllowed := []string{organizationKey, clusterGroupKey, clusterKey, workspaceKey, namespaceKey}
+
+	if v, ok := scopeData[organizationKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, organizationKey)
+		}
+	}
+
+	if v, ok := scopeData[clusterGroupKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterGroupKey)
+		}
+	}
+
+	if v, ok := scopeData[clusterKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterKey)
+		}
+	}
+
+	if v, ok := scopeData[workspaceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, workspaceKey)
+		}
+	}
+
+	if v, ok := scopeData[namespaceKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, namespaceKey)
+		}
+	}
+
+	if len(scopesFound) == 0 {
+		return fmt.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v", scopesAllowed)
+	} else if len(scopesFound) > 1 {
+		return fmt.Errorf("found scopes: %v are not valid: maximum one valid scope type block is allowed", scopesFound)
+	}
```

Comment:
Add the delimiter `,`

```suggestion
	if len(scopesFound) == 0 {
		return fmt.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v", strings.Join(scopesAllowed, `, `))
	} else if len(scopesFound) > 1 {
		return fmt.Errorf("found scopes: %v are not valid: maximum one valid scope type block is allowed", strings.Join(scopesFound, `, `))
	}
```